packages/@aws-cdk/aws-apigateway/test/stepfunctions-api.test.ts

```diff
@@ -0,0 +1,419 @@
+              ],
+            ],
+          },
+        },
+        Type: 'AWS',
+        Uri: {
+          'Fn::Join': [
+            '',
+            [
+              'arn:',
+              {
+                Ref: 'AWS::Partition',
+              },
+              ':apigateway:',
+              {
+                Ref: 'AWS::Region',
+              },
+              ':states:action/StartSyncExecution',
+            ],
+          ],
+        },
+        PassthroughBehavior: 'NEVER',
+      },
+    });
+  });
+
+  test('StepFunctionsRestApi defines correct REST API resources with includeRequestContext set to true', () => {
+    //GIVEN
+    const { stack, stateMachine } = givenSetup();
+
+    //WHEN
+    const api = new apigw.StepFunctionsRestApi(stack,
+      'StepFunctionsRestApi', {
+        stateMachine: stateMachine,
+        requestContext: {
+          accountId: true,
+          apiId: true,
+          apiKey: true,
+          authorizerPrincipalId: true,
+          caller: true,
+          cognitoAuthenticationProvider: true,
+          cognitoAuthenticationType: true,
+          cognitoIdentityId: true,
+          cognitoIdentityPoolId: true,
+          httpMethod: true,
+          stage: true,
+          sourceIp: true,
+          user: true,
+          userAgent: true,
+          userArn: true,
+          requestId: true,
+          resourceId: true,
+          resourcePath: true,
+        },
+      });
+
+    expect(() => {
+      api.root.addResource('not allowed');
+    }).toThrow();
+
+    //THEN
+    expect(stack).toHaveResource('AWS::ApiGateway::Method', {
```

Comment:
As mentioned in the above comment, whittle down the assertion to the property that is impacted and use matchers like `stringLike()` to focus on the most important part of the assertion.